tsdb/querier.go

```diff
@@ -79,11 +79,13 @@ func newBlockBaseQuerier(b BlockReader, mint, maxt int64) (*blockBaseQuerier, er
 
 func (q *blockBaseQuerier) LabelValues(ctx context.Context, name string, hints *storage.LabelHints, matchers ...*labels.Matcher) ([]string, annotations.Annotations, error) {
 	res, err := q.index.SortedLabelValues(ctx, name, matchers...)
```

Comment:
I'm wondering if `hints` should be passed down to `q.index.SortedLabelValues`? If so, we could improve performance by not fetching more values than necessary (instead of truncating after the fact).